src/View/Helper/CalendarHelper.php

```diff
@@ -296,7 +296,17 @@ public function previousLink() {
 			$prevMonth = $prevYear = null;
 		}
 
-		return $this->Html->link(__('previous'), [$prevYear, $this->formatMonth($prevMonth)]);
+		$url = [
+			$prevYear,
+			$this->formatMonth($prevMonth),
+		];
+
+		$viewVars = $this->_View->get('_calendar');
+		if (!empty($viewVars['url'])) {
+			$url = array_merge($url, $viewVars['url']);
```

Comment:
Why merging and not just replacing?
The merging sounds like it could go wrong in some cases.

Reply:
I don't want to require the userland code to set the year / month variables.

Reply:
> What if there are query strings on the current URL, how would the new URL query strings get merged in?

The existing implementation doesn't respect query strings from the current URL; that's what I'm trying to replicate.